src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -1221,3 +1288,31 @@ int sd_dhcp_server_set_callback(sd_dhcp_server *server, sd_dhcp_server_callback_
 
         return 0;
 }
+
+int sd_dhcp_server_add_static_lease(sd_dhcp_server *server, sd_dhcp_static_lease *v) {
+        _cleanup_(sd_dhcp_static_lease_unrefp) sd_dhcp_static_lease *old = NULL;
+        int r;
+
+        assert_return(server, -EINVAL);
+        assert_return(v, -EINVAL);
+
+        r = hashmap_ensure_allocated(&server->static_leases_by_client_id, &dhcp_static_lease_hash_ops);
+        if (r < 0)
+                return r;
+
+        /* Static lease with an empty or omitted address is a valid entry,
+         * it removes any static lease with the specified mac address. */
```

Comment:
not a fan of this. why not add `sd_dhdpc_server_remove_static_lease()`?

not a fan of overloading add functions as removal functions i must say.